mlflow/data/tensorflow_dataset.py

```diff
@@ -102,83 +127,120 @@ def profile(self) -> Optional[Any]:
         """
         A profile of the dataset. May be None if no profile is available.
         """
-        import tensorflow as tf
-
         profile = {
-            "features_num_rows": len(self._data),
```

Comment:
`num_rows` doesn't quite make sense for tensors, since tensors don't have rows. I think cardinality is good enough.